### ISSUE TITLE
[BUG] build failed use winlibs-x86_64-posix-seh-gcc-14.2.0-llvm-18.1.8-mingw-w64ucrt-12.0.0-r1

### ISSUE THREAD

**Issue by wildtang** [notepad-plus-plus/notepad-plus-plus#15581]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Issue

build failed use winlibs-x86_64-posix-seh-gcc-14.2.0-llvm-18.1.8-mingw-w64ucrt-12.0.0-r1
Below is error info:

> E:\Research\GitHub\notepad-plus-plus>git checkout v8.6.9
HEAD is now at 76106081 Notepad++ v8.6.9 release

> E:\Research\GitHub\notepad-plus-plus>cd PowerEditor\gcc

> E:\Research\GitHub\notepad-plus-plus\PowerEditor\gcc>mingw32-make
process_begin: CreateProcess(NULL, echo + creating BUILD_DIRECTORY bin.x86_64.build, ...) failed.
make (e=2): 系统找不到指定的文件。
mingw32-make: *** [makefile:237: bin.x86_64.build] Error 2

### Steps To Reproduce

1. E:\Research\GitHub\notepad-plus-plus>git checkout v8.6.9 
2. git checkout v8.6.9
3. cd PowerEditor\gcc 
4. mingw32-make 

### Current Behavior

E:\Research\GitHub\notepad-plus-plus\PowerEditor\gcc>mingw32-make

process_begin: CreateProcess(NULL, echo + creating BUILD_DIRECTORY bin.x86_64.build, ...) failed.
make (e=2): 系统找不到指定的文件。(means file not found)
mingw32-make: *** [makefile:237: bin.x86_64.build] Error 2

### Expected Behavior

Build finished without errors.

### Debug Information

```shell
Nope. (Build error.)
```

**Comment by rdipardo** [comment]
The only significant difference is the compiler (*hint: the source tree path is /PowerEditor/&#x200D;__gcc__*).

The code is simply incompatible with Clang's default warning settings, which tend to be more pedantic than GCC's. Any version of Clang will run into similar problems, e.g.,

~~~sh-session
$ clang++ --version
clang version 17.0.3
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\Llvm\x64\bin

$ set "CXX=clang++" & mingw32-make -f makefile
+ creating BUILD_DIRECTORY bin.x86_64.build
+ creating SCINTILLA_BUILD_DIRECTORY bin.x86_64.build/_scintilla.build
[... snip ...]
+ compiling ../src/EncodingMapper.cpp
+ compiling ../src/DarkMode/DarkMode.cpp
+ compiling ../src/MISC/Exception/MiniDumper.cpp
+ compiling ../src/MISC/Common/FileInterface.cpp
+ compiling ../src/MISC/Exception/Win32Exception.cpp
+ compiling ../src/MISC/Common/verifySignedfile.cpp
+ compiling ../src/MISC/Common/Common.cpp
+ compiling ../src/MISC/Process/Processus.cpp
+ compiling ../src/MISC/PluginsManager/PluginsManager.cpp
+ compiling ../src/MISC/RegExt/regExtDlg.cpp
+ compiling ../src/MISC/sha2/sha-256.cpp
+ compiling ../src/MISC/sha1/calc_sha1.cpp
+ compiling ../src/MISC/sha512/sha512.cpp
+ compiling ../src/MISC/sha1/sha1.cpp
+ compiling ../src/MISC/md5/md5Dlgs.cpp
+ compiling ../src/Notepad_plus_Window.cpp
+ compiling ../src/NppBigSwitch.cpp
+ compiling ../src/NppCommands.cpp
+ compiling ../src/Notepad_plus.cpp
+ compiling ../src/NppDarkMode.cpp
+ compiling ../src/NppNotification.cpp
+ compiling ../src/NppIO.cpp
+ compiling ../src/ScintillaComponent/AutoCompletion.cpp
+ compiling ../src/ScintillaComponent/Buffer.cpp
../src/ScintillaComponent/Buffer.cpp:1731:68: error: '_exception_code' only allowed in __except block or filter expression
 1731 |                                 _stprintf_s(szException, _countof(szException), L"0x%X (SEH)", ::GetExceptionCode());
      |                                                                                                ^
1 error generated.
mingw32-make[1]: *** [makefile:264: bin.x86_64.build/ScintillaComponent/Buffer.o] Error 1
mingw32-make[1]: *** Waiting for unfinished jobs....
+ compiling ../src/ScintillaComponent/DocTabView.cpp
+ compiling ../src/Parameters.cpp
+ compiling ../src/ScintillaComponent/FindReplaceDlg.cpp
mingw32-make: *** [makefile:234: all] Error 2
~~~

**Comment by rdipardo** [comment]
Wait a minute, the specific target that failed was this:

https://github.com/notepad-plus-plus/notepad-plus-plus/blob/bdf3ccaca91619b803c09a6615a7bf2553ee00a2/PowerEditor/gcc/makefile#L237-L238

And the error message was "系统找不到指定的文件", or "The system cannot find the file specified".

So, @wildtang, it looks more like `mkdir.exe` is not in the `PATH` of your Windows system for some reason. The MYSYS2 shell creates it's own environment, so everything should work there.

In other words, this looks like a misconfigured system `PATH`, not a Notepad++ code issue, and not even a Clang issue &mdash; even if Clang does appear to cause some other, unrelated issues.

**Comment by wildtang** [comment]
I checked, not the problem of mkdir not in the PATH.

After download and unpack winlibs-x86_64-posix-seh-gcc-14.2.0-llvm-18.1.8-mingw-w64ucrt-12.0.0-r1.zip,
I write a bat to launch winlibs.
![图片](https://github.com/user-attachments/assets/d4da0374-f2ee-40a4-95a8-5fddd86b2d40)

And I can confirm mkdir command works.
![图片](https://github.com/user-attachments/assets/46bd2a81-a19c-4743-bc3d-35c469c53dd4)

![图片](https://github.com/user-attachments/assets/9e2ac400-8253-4c0f-86aa-542043ad6e2c)

**Comment by rdipardo** [comment]
@wildwang

O.K. you are right. The actually problem is clear if you expand all the tasks using the `-n` flag:

~~~sh-session
$ clang++ --version
(built by Brecht Sanders, r1) clang version 18.1.8
Target: x86_64-w64-windows-gnu
Thread model: posix
InstalledDir: C:/Users/Rob/scoop/apps/winlibs-mingw-llvm-ucrt/current/bin

$ set "CXX=clang++" & mingw32-make -n
echo + creating BUILD_DIRECTORY bin.x86_64.build
mkdir bin.x86_64.build bin.x86_64.build\DarkMode bin.x86_64.build\MISC\Common bin.x86_64.build\MISC\Exception bin.x86_64.build\MISC\PluginsManager bin.x86_64.build\MISC\Process bin.x86_64.build\MISC\RegExt bin.x86_64.build\MISC\md5 bin.x86_64.build\MISC\sha1 bin.x86_64.build\MISC\sha2 bin.x86_64.build\MISC\sha512 bin.x86_64.build\ScintillaComponent bin.x86_64.build\TinyXml bin.x86_64.build\TinyXml\tinyXmlA bin.x86_64.build\WinControls\AboutDlg bin.x86_64.build\WinControls\AnsiCharPanel bin.x86_64.build\WinControls\ClipboardHistory bin.x86_64.build\WinControls\ColourPicker bin.x86_64.build\WinControls\ContextMenu bin.x86_64.build\WinControls\DockingWnd bin.x86_64.build\WinControls\DocumentMap bin.x86_64.build\WinControls\DoubleBuffer bin.x86_64.build\WinControls\FileBrowser bin.x86_64.build\WinControls\FindCharsInRange bin.x86_64.build\WinControls\Grid bin.x86_64.build\WinControls\ImageListSet bin.x86_64.build\WinControls\OpenSaveFileDialog bin.x86_64.build\WinControls\PluginsAdmin bin.x86_64.build\WinControls\Preference bin.x86_64.build\WinControls\ProjectPanel bin.x86_64.build\WinControls\ReadDirectoryChanges bin.x86_64.build\WinControls\SplitterContainer bin.x86_64.build\WinControls\StaticDialog bin.x86_64.build\WinControls\StaticDialog\RunDlg bin.x86_64.build\WinControls\StatusBar bin.x86_64.build\WinControls\TabBar bin.x86_64.build\WinControls\TaskList bin.x86_64.build\WinControls\ToolBar bin.x86_64.build\WinControls\ToolTip bin.x86_64.build\WinControls\TrayIcon bin.x86_64.build\WinControls\TreeView bin.x86_64.build\WinControls\VerticalFileSwitcher bin.x86_64.build\WinControls\WindowsDlg bin.x86_64.build\WinControls\functionList bin.x86_64.build\WinControls\shortcut bin.x86_64.build\uchardet [...]

~~~

Windows `mkdir` does not create subdirectories by default and does not support the `-p` flag, so the flag is not set for Command Prompt sessions, only for MSYS2 or PowerShell (because the `ls` command exists there).

https://github.com/notepad-plus-plus/notepad-plus-plus/blob/bdf3ccaca91619b803c09a6615a7bf2553ee00a2/PowerEditor/gcc/makefile#L200-L201

@ArkadiuszMichalski

Standard DOS `mkdir` actually can make directory trees, but there must be a slash (`\`) before the root folder name: <https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/mkdir#examples>

Is it worth updating the Makefile just for Command Prompt sessions? Not sure.

**Comment by wildtang** [comment]
@rdipardo Thanks for your very enthusiastic replies👏

I resolved by add a git bin folder into PATH list before any other PATH to the console bat for WinLibs I made as below.(the git bin folder contain mkdir.exe and rmdir.exe), Now mingw32-make and mingw32-make clean both works well.

```
SET PATH=E:\Research\WinLibs\Current\bin;C:\Users\wild\scoop\apps\git\2.46.0\usr\bin;%PATH%
cmd

```
